File: app/models/assessment/LanguageInspection.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.assessment;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import java.util.Date;
import models.base.OwnedModel;
import models.exam.Exam;
import models.user.User;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

@Entity
public class LanguageInspection extends OwnedModel {

    @OneToOne
    private Exam exam;

    @ManyToOne
    private User assignee;

    @OneToOne
    private Comment statement;

    @Temporal(TemporalType.TIMESTAMP)
    private Date startedAt;

    @Temporal(TemporalType.TIMESTAMP)
    private Date finishedAt;

    private Boolean approved;

    public Exam getExam() {
        return exam;
    }

    public void setExam(Exam exam) {
        this.exam = exam;
    }

    public Boolean getApproved() {
        return approved;
    }

    public void setApproved(Boolean approved) {
        this.approved = approved;
    }

    public Date getFinishedAt() {
        return finishedAt;
    }

    public void setFinishedAt(Date finishedAt) {
        this.finishedAt = finishedAt;
    }

    public Date getStartedAt() {
        return startedAt;
    }

    public void setStartedAt(Date startedAt) {
        this.startedAt = startedAt;
    }

    public Comment getStatement() {
        return statement;
    }

    public void setStatement(Comment statement) {
        this.statement = statement;
    }

    public User getAssignee() {
        return assignee;
    }

    public void setAssignee(User assignee) {
        this.assignee = assignee;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (!(o instanceof LanguageInspection that)) return false;

        return new EqualsBuilder().append(id, that.id).isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(17, 37).append(id).toHashCode();
    }
}

File: app/models/assessment/LanguageInspection.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.assessment

import jakarta.persistence.*
import models.base.OwnedModel
import models.exam.Exam
import models.user.User

import java.util.Date
import scala.compiletime.uninitialized

@Entity
class LanguageInspection extends OwnedModel:
  @OneToOne
  var exam: Exam = uninitialized

  @ManyToOne
  var assignee: User = uninitialized

  @OneToOne
  var statement: Comment = uninitialized

  @Temporal(TemporalType.TIMESTAMP)
  var startedAt: Date = uninitialized

  @Temporal(TemporalType.TIMESTAMP)
  var finishedAt: Date = uninitialized

  var approved: java.lang.Boolean = uninitialized

  override def equals(o: Any): Boolean = o match
    case l: LanguageInspection => this.id == l.id
    case _                     => false

  override def hashCode: Int = id.toInt
